File: backend/app/models/budget.py
from datetime import datetime
from sqlmodel import Field, SQLModel
from typing import Optional

class Budget(SQLModel, table=True):
    __tablename__ = "budgets"

    id: int | None = Field(default=None, primary_key=True)
    user_id: int = Field(foreign_key="users.id")
    amount: float = Field(gt=0)
    category: str | None = None  # If null, it's the total budget for the month
    month: str  # Format: "YYYY-MM"
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: datetime = Field(default_factory=datetime.now)

File: android/app/src/main/java/com/ledger/ledgerapp/network/models/Budget.kt
package com.ledger.ledgerapp.network.models

import com.google.gson.annotations.SerializedName

data class Budget(
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    val amount: Double,
    val category: String?,
    val month: String,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("updated_at")
    val updatedAt: String
)

data class BudgetRequest(
    val amount: Double,
    val category: String?,
    val month: String
)
